vulnerabilities/import_runner.py

```diff
@@ -32,65 +32,34 @@
 
 
 class ImportRunner:
     """
     The ImportRunner is responsible for inserting and updating data about vulnerabilities and
-    affected/unaffected/fixed packages in the database. The two main goals for the implementation
-    are correctness and efficiency.
+    affected/unaffected/fixed packages in the database. The main goal for the implementation
+    is correctness
 
     Correctness:
         - There must be no duplicates in the database (should be enforced by the schema).
         - No valid data from the data source must be skipped or truncated.
-
-    Efficiency:
-        - Bulk inserts should be used whenever possible.
-        - Checking whether a record already exists should be kept to a minimum
-        (the data source should know this instead).
-        - All update and select operations must use indexed columns.
     """
 
-    def __init__(self, importer: models.Importer):
+    def __init__(self, importer: DataSource):
         self.importer = importer
 
-    def run(self, cutoff_date: datetime.datetime = None) -> None:
+    def run(self) -> None:
         """
         Create a data source for the given importer and store the data retrieved in the database.
-
-        cutoff_date - optional timestamp of the oldest data to include in the import
-
-        NB: Data sources provide two kinds of records; vulnerabilities and packages. Vulnerabilities
-        are potentially shared across many packages, from the same data source and from different
-        data sources. For example, a vulnerability in the Linux kernel is mentioned by advisories
-        from all Linux distributions that package this kernel version.
         """
-        logger.info(f"Starting import for {self.importer.name}.")
-        data_source = self.importer.make_data_source(cutoff_date=cutoff_date)
-        with data_source:
-            advisory_data = data_source.advisory_data()
-            importer_name = data_source.qualified_name()
-            process_advisories(advisory_datas=advisory_data, importer_name=importer_name)
-        self.importer.last_run = datetime.datetime.now(tz=datetime.timezone.utc)
-        self.importer.data_source_cfg = dataclasses.asdict(data_source.config)
-        self.importer.save()
-
-        logger.info(f"Finished import for {self.importer.name}.")
-
-
-def vuln_ref_exists(vulnerability, url, reference_id):
-    return models.VulnerabilityReference.objects.filter(
-        vulnerability=vulnerability, reference_id=reference_id, url=url
-    ).exists()
-
-
-def get_vuln_pkg_refs(vulnerability, package):
-    return models.PackageRelatedVulnerability.objects.filter(
-        vulnerability=vulnerability,
-        package=package,
-    )
+        logger.info(f"Starting import for {self.importer.qualified_name}")
+        advisory_datas = self.importer().advisory_data()
+        importer_name = self.importer.qualified_name
+        process_advisories(advisory_datas=advisory_datas, importer_name=importer_name)
```

Comment:
This would be best returning something IMHO, possibly yielding what the key for inserted records?

Reply:
This sounds a little off to me as `process_advisories` is a function that is supposed to process all the advisories and be done with it. It is not required to run as you consume the output (in case of generators).

I've implemented it anyway.

Reply:
For now, I've made it to return the total count of imported advisories